flax2bnn.py

```diff
@@ -0,0 +1,157 @@
+"""Convert frequentist Flax modules to Bayesian NNs."""
+import operator
+from functools import reduce
+from typing import Optional
+
+import chex
+import flax.linen as nn
+import jax.numpy as jnp
+import numpyro.distributions as dist # ! Rather use jax.scipy distributions to avoid extra dependencies
+
+
+def get_by_path(root: dict, items: list):
+    """Access a nested object in root by item sequence."""
+    return reduce(operator.getitem, items, root)
+
+
+def get_flattened_keys(d: dict, sep='.'):
+    """
+    Recursively get & concat the keys of a dictionary and its subdictionaries.
+
+    Args:
+    d: dict - dictionary to get the keys from
+    sep: str - separator for the keys (default: '.')
+
+    Returns:
+    list: list of keys
+    """
+    keys = []
+    for k, v in d.items():
+        if isinstance(v, dict):
+            keys.extend([f'{k}{sep}{kk}' for kk in get_flattened_keys(v)])
+        else:
+            keys.append(k)
+    return keys
+
+
+class ProbModelBuilder:
+    """Convert frequentist Flax modules to Bayesian Numpyro modules."""
+
+    def __init__(
+        self,
+        module: nn.Module,
+        prior_config: dict,
+        params: dict,
+        seed: int,
+        evaluate: bool = False,
+        full_batch_len: Optional[int] = None,
+    ):
+        """Initialize the model builder."""
+        self.module = module
+        self.prior_config = prior_config
+        self.params = params
+        self.seed = seed
+        self.evaluate = evaluate
+        self.full_batch_len = full_batch_len if full_batch_len is not None else 1
+        self.priorsdict = self.set_prior()
+
+    def map_distribution(self, config: dict) -> dist.Distribution:
+        """Map the distribution from string to callable."""
+        if config['dist'] == 'Normal':
+            return dist.Normal(scale=config['sd'])
+        elif config['dist'] == 'Laplace':
+            return dist.Laplace(scale=config['sd'])
+        else:
+            raise ValueError(f'Distribution {config["dist"]} not implemented')
```

Comment:
A bit of a pet peeve, but I think it would slightly improve readability
```suggestion
    def map_distribution(self, config: dict) -> dist.Distribution:
        """Map the distribution from string to callable."""
        if config['dist'] == 'Normal':
            return dist.Normal(scale=config['sd'])
        if config['dist'] == 'Laplace':
            return dist.Laplace(scale=config['sd'])
        raise ValueError(f'Distribution {config["dist"]} not implemented')
```

Reply:
good one!